evalml/tests/automl_tests/test_automl.py

```diff
@@ -585,6 +585,31 @@ def test_large_dataset_regression(mock_score):
         assert automl.results['pipeline_results'][pipeline_id]['cv_data'][0]['score'] == 1.234
 
 
+@patch('evalml.pipelines.RegressionPipeline.score')
+def test_large_dataset_split_size(mock_score):
+    def generate_fake_dataset(rows):
+        X = pd.DataFrame({'col_0': [i for i in range(rows)]})
+        y = pd.Series([i % 2 for i in range(rows)])
+        return X, y
+
+    fraud_objective = FraudCost(amount_col='col_0')
+
+    automl = AutoMLSearch(problem_type='binary',
+                          objective=fraud_objective,
+                          additional_objectives=['auc', 'f1', 'precision'],
+                          max_time=1,
+                          max_pipelines=1,
+                          optimize_thresholds=True)
+    mock_score.return_value = {automl.objective.name: 1.234}
+    assert automl.data_split is None
+
+    under_max_rows = automl._LARGE_DATA_ROW_THRESHOLD + 1
+    X, y = generate_fake_dataset(under_max_rows)
+    automl.search(X, y)
+    assert isinstance(automl.data_split, TrainingValidationSplit)
```

Comment:
This is great. Do we have coverage where we check that if we're under the max row threshold, the result is a `CrossValidationSplit` ? If not, let's add it.